### PR TITLE
[Refactor] Move transaction methods to another file and more tests

### PR DESCRIPTION
**Pull request by sduzh**
- Move lake::TabletManager::{publish_version(), abort_txn(), publish_log_version()} to another file.
- Add some unit tests to cover more cases.
- Use the higher-level LakeService module's publish_version/abort_txn/publish_log_version interface in unit tests.
- Reducing Duplicate Code in Unit Tests.
- Remove two unused methods: lake::TabletManager::{delete_segment(), delete_del()}

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 110 / 131 (83.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/service/service_be/starrocks_be.cpp|0|1|00.00%|[205]|
|:large_blue_circle:|src/storage/lake/transactions.cpp|96|114|84.21%|[31, 57, 103, 104, 121, 125, 126, 131, 132, 166, 167, 168, 171, 186, 196, 205, 206, 207]|
|:large_blue_circle:|src/service/service_be/lake_service.cpp|14|16|87.50%|[398, 748]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 110 / 131 (83.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/service/service_be/starrocks_be.cpp|0|1|00.00%|[205]|
|:large_blue_circle:|src/storage/lake/transactions.cpp|96|114|84.21%|[31, 57, 103, 104, 121, 125, 126, 131, 132, 166, 167, 168, 171, 186, 196, 205, 206, 207]|
|:large_blue_circle:|src/service/service_be/lake_service.cpp|14|16|87.50%|[398, 748]|

File: be/src/storage/lake/tablet_manager.cpp [hunk 1]
```diff
@@ -124,10 +123,6 @@ static bvar::PassiveStatus<size_t> g_metacache_capacity("lake", "metacache_capac
 static bvar::PassiveStatus<size_t> g_metacache_usage("lake", "metacache_usage", get_metacache_usage, nullptr);
 #endif
 
-static StatusOr<TabletMetadataPtr> publish(TabletManager* tablet_mgr, Tablet* tablet, int64_t base_version,
```
Comment (sduzh) [inline be/src/storage/lake/tablet_manager.cpp:127]: This function has been moved to the new file transactions.{h, cpp}

File: be/src/storage/lake/tablet_manager.cpp [hunk 2]
```diff
@@ -557,18 +557,6 @@ Status TabletManager::delete_txn_vlog(int64_t tablet_id, int64_t version) {
     return st.is_not_found() ? Status::OK() : st;
 }
 
-Status TabletManager::delete_segment(int64_t tablet_id, std::string_view segment_name) {
```
Comment (sduzh) [inline be/src/storage/lake/tablet_manager.cpp:560]: Unused method, deleted

File: be/src/storage/lake/tablet_manager.cpp [hunk 3]
```diff
@@ -557,18 +557,6 @@ Status TabletManager::delete_txn_vlog(int64_t tablet_id, int64_t version) {
...
-    erase_metacache(segment_name);
-    auto st = fs::delete_file(segment_location(tablet_id, segment_name));
-    return st.is_not_found() ? Status::OK() : st;
-}
-
-Status TabletManager::delete_del(int64_t tablet_id, std::string_view del_name) {
```
Comment (sduzh) [inline be/src/storage/lake/tablet_manager.cpp:566]: Unused method, deleted

File: be/src/storage/lake/tablet_manager.cpp [hunk 4]
```diff
@@ -679,139 +667,6 @@ StatusOr<TabletSchemaPtr> TabletManager::get_tablet_schema_by_index_id(int64_t t
     }
 }
 
-StatusOr<TabletMetadataPtr> TabletManager::publish_version(int64_t tablet_id, int64_t base_version, int64_t new_version,
-                                                           const int64_t* txns, int txns_size, int64_t commit_time) {
-    ASSIGN_OR_RETURN(auto tablet, get_tablet(tablet_id));
-    return publish(this, &tablet, base_version, new_version, txns, txns_size, commit_time);
-}
-
-StatusOr<TabletMetadataPtr> publish(TabletManager* tablet_mgr, Tablet* tablet, int64_t base_version,
```
Comment (sduzh) [inline be/src/storage/lake/tablet_manager.cpp:688]: Moved to transactions.cpp

File: be/src/storage/lake/tablet_manager.cpp [hunk 5]
```diff
@@ -827,72 +682,6 @@ StatusOr<CompactionTaskPtr> TabletManager::compact(int64_t tablet_id, int64_t ve
     }
 }
 
-void TabletManager::abort_txn(int64_t tablet_id, const int64_t* txns, int txns_size) {
```
Comment (sduzh) [inline be/src/storage/lake/tablet_manager.cpp:830]: Moved to transactions.cpp

File: be/src/storage/lake/tablet_manager.cpp [hunk 6]
```diff
@@ -827,72 +682,6 @@ StatusOr<CompactionTaskPtr> TabletManager::compact(int64_t tablet_id, int64_t ve
...
-
-        files_to_delete.emplace_back(log_path);
-
-        metacache()->erase(CacheKey(log_path));
-    }
-
-    delete_files_async(std::move(files_to_delete));
-}
-
-Status TabletManager::publish_log_version(int64_t tablet_id, int64_t txn_id, int64 log_version) {
```
Comment (sduzh) [inline be/src/storage/lake/tablet_manager.cpp:872]: Moved to transactions.cpp

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 131 / 139 (94.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/service/service_be/starrocks_be.cpp|0|1|00.00%|[205]|
|:large_blue_circle:|src/service/service_be/lake_service.cpp|16|18|88.89%|[399, 749]|
|:large_blue_circle:|src/storage/lake/transactions.cpp|109|114|95.61%|[31, 57, 167, 186, 196]|
|:large_blue_circle:|src/storage/lake/tablet_manager.cpp|3|3|100.00%|[]|
|:large_blue_circle:|src/fs/fs_util.h|1|1|100.00%|[]|
|:large_blue_circle:|src/storage/lake/txn_log_applier.cpp|2|2|100.00%|[]|

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.2

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**